### PR TITLE
Project refactoring

### PR DESCRIPTION
**Pull request by st58238**
Janáček Lukáš (st58238)

### CODE REVIEW

File: NNPTPZ1/ComplexNumber.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
+﻿using System;
+
+namespace NNPTPZ1.Mathematics
+{
+    public class ComplexNumber
+    {
+        public double RealPart { get; set; }
+        public float ImaginaryPart { get; set; }
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: why float, if real is double?

File: NNPTPZ1/ComplexNumber.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            if (obj is ComplexNumber)
+            {
+                ComplexNumber x = obj as ComplexNumber;
+                return x.RealPart == RealPart && x.ImaginaryPart == ImaginaryPart;
+            }
+            return base.Equals(obj);
+        }
+
+        public readonly static ComplexNumber Zero = new ComplexNumber()
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs:10]: properties should be at the top of class

File: NNPTPZ1/ComplexNumber.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            ComplexNumber a = this;
+            return new ComplexNumber()
+            {
+                RealPart = a.RealPart + b.RealPart,
+                ImaginaryPart = a.ImaginaryPart + b.ImaginaryPart
+            };
+        }
+
+        public double GetAngleInDegrees()
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: really degrees?

File: NNPTPZ1/ComplexNumber.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            return base.Equals(obj);
+        }
+
+        public readonly static ComplexNumber Zero = new ComplexNumber()
+        {
+            RealPart = 0,
+            ImaginaryPart = 0
+        };
+
+        public ComplexNumber Multiply(ComplexNumber b)
```
Comment (rodi0878) [inline NNPTPZ1/ComplexNumber.cs]: reorganize file, math operations should be close together

File: NNPTPZ1/NewtonFractalGenerator.cs [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+                {
+                    double x = XMin + i * xStep;
+                    double y = YMin + j * yStep;
+
+                    ComplexNumber ox = new ComplexNumber()
+                    {
+                        RealPart = (x == 0 ? 0.001 : x),
+                        ImaginaryPart = (y == 0 ? 0.001f : (float)y)
+                    };
+                    Colorize(i, j, FindRoots(ox), Iterate(ox));
```
Comment (rodi0878) [inline NNPTPZ1/NewtonFractalGenerator.cs]: this changes the original logic of algorithm!

File: NNPTPZ1/NewtonFractalGenerator.cs [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+            CheckIntParsable(arguments[0], "Argument on position 1 is not a valid positive integer");
+            CheckIntParsable(arguments[1], "Argument on position 2 is not a valid positive integer");
+
+            CheckDoubleParsable(arguments[2], "Argument on position 3 is not a valid number");
+            CheckDoubleParsable(arguments[3], "Argument on position 4 is not a valid number");
+            CheckDoubleParsable(arguments[4], "Argument on position 5 is not a valid number");
+            CheckDoubleParsable(arguments[5], "Argument on position 6 is not a valid number");
+
+            using (FileStream test = File.OpenWrite(arguments[6] ?? "../../../out.png")) { }
```
Comment (rodi0878) [inline NNPTPZ1/NewtonFractalGenerator.cs]: useless

**Review by rodi0878 — CHANGES_REQUESTED** [review]
Please review comments and provide update

**Comment by st58238** [comment]
Implemented requested changes.

**Comment by rodi0878** [comment]
accepted

**Pull request closed without merging**